src/classes/window/fullscreen.vala

```diff
@@ -182,7 +182,7 @@ namespace pdfpc.Window {
             var monitor_geometry = this.monitor.get_geometry();
             this.window_w = monitor_geometry.width;
             this.window_h = monitor_geometry.height;
-            if (Pdfpc.is_Wayland_backend() && Options.wayland_workaround) {
+            if (Pdfpc.is_Wayland_backend() || Pdfpc.is_Quartz_backend() && Options.wayland_workaround) {
```

Comment:
Are you sure this is needed for mac? In any case, this certainly breaks the intended Wayland-related functionality.

Reply:
Hmm, I forgot to add parentheses.

Reply:
> Hmm, I forgot to add parentheses.

Right, but the main question is whether such a workaround is needed under macOS in the first place. If yes, it should probably be renamed.

Reply:
If I remember correctly, when I ran without `-W` it didn't fix the scaling. But let me double check tomorrow. I'll get back to you tomorrow unless the university decides to close abruptly.